test/test_backend/routes/tst_graph.py

```diff
@@ -32,70 +38,100 @@
-            print()
-
-
+        self.setUpClass()  # temp until tests finished
+        for test_type, test_name, codeset_ids, timeout_secs in [x.values() for x in self.test_cases]:
+            # TODO: temp
+            if test_name not in ['antibiotics 1']:
+                continue
+            input_path = THIS_INPUT_DIR / CONNECTED_SUBGRAPH_FILENAME_PATTERN.format(test_name)
+            sg, nodes_in_graph, preferred_concept_ids, orphans_not_in_graph, hidden = \
+                get_connected_subgraph(REL_GRAPH, codeset_ids, EXTRA_CONCEPT_IDS, HIDE_VOCABS)
+            #
+            normalized_hidden = {k: sorted(list(v)) for k, v in hidden.items()}
+            results = {
+                'sg.edges': sorted(list(sg.edges)),
+                'nodes_in_graph': sorted(list(nodes_in_graph)),
+                'preferred_concept_ids': sorted(list(preferred_concept_ids)),
+                'orphans_not_in_graph': sorted(list(orphans_not_in_graph)),
+                'hidden': normalized_hidden,
+            }
+            with open(input_path, 'w') as f:
+                f.write(json.dumps(results, indent=4))
+
+    async def tst_get_connected_subgraph(self, only_fast_cases=True):
+        """Test indented_concept_list()"""
+        self.setUpClass()  # temp until tests finished
+        for test_type, test_name, codeset_ids, timeout_secs in [x.values() for x in self.test_cases]:
+            # Do just 1 fast test case, or all test cases?
+            if only_fast_cases and test_name not in ['many-small', 'single-small', 'single-small-second-time']:
+                continue
+            # Expected
+            input_path = THIS_INPUT_DIR / CONNECTED_SUBGRAPH_FILENAME_PATTERN.format(test_name)
+            expected_raw = json.load(open(input_path, 'r'))
+            expected = {}  # convert JSON serialization back to sets
+            for k, v in expected_raw.items():
+                if k == 'sg.edges':
+                    expected[k] = {tuple(x) for x in v}
+                elif k == 'hidden':
+                    expected[k] = {k2: set(v2) for k2, v2 in v.items()}
+                else:
+                    expected[k] = set(v)
+            # Actual
+            sg, nodes_in_graph, preferred_concept_ids, orphans_not_in_graph, hidden = \
+                get_connected_subgraph(REL_GRAPH, codeset_ids, EXTRA_CONCEPT_IDS, HIDE_VOCABS)
+            actual = {
+                'sg.edges': set(sg.edges),
+                'nodes_in_graph': nodes_in_graph,
+                'preferred_concept_ids': preferred_concept_ids,
+                'orphans_not_in_graph': orphans_not_in_graph,
+                'hidden': hidden,
+            }
+            # Compare
+            for k, v in actual.items():
+                # self.assertEqual(v, expected[k], msg=f'Results differ for {k} for test {test_name}')
+                assert v == expected[k], f'Results differ for "{k}" in test for "{test_name}"'
```

Comment:
## Test results
I made 3 PRs:
- #663 - This is the commit you wanted me to start from. This is where I set the base case. I ran the results and saved the outputs into an `input/` folder. These results differed significantly than what you had in your `results` variable in the Python code you gave me and, as discussed in another comment, I took it that those results were just a placeholder and that I should not worry about this.
- #664 - This is the only other commit other than the current state of `develop` that you asked me to test against.
- #655 - This one; the only one that will be merged.

Pass/fail results:
- #663 - Created input files. Of course, tests passed since it was comparing results to the output it had just saved.
- #644 - Passed!
- #655 - Failed. See below.

Failures
So the only failures were for #655. I got 1 failure for each of the "`only_fast_cases`" that I tested, `['many-small', 'single-small', 'single-small-second-time']`. The only failed when comparing against `sg.edges`, and the set differences I got back comparing actual vs expected were the same for all 3:

<details><summary>Set differences</summary>

**`expected.difference(actual)`**
`set()`

**`actual.difference(expected)`**
```
{(437515, 4130161), (441840, 4041283), (4214376, 4034966), (4210128, 4161671), (4274025, 432795), (37311253, 35626038), (35622958, 45766050), (4274025, 4177703), (45757798, 380096), (441840, 255919), (4034964, 37311673), (761191, 35625722), (376683, 37311253), (4082346, 37110593), (4029423, 4016047), (37311254, 35626039), (436670, 437515), (35622016, 4129525), (4174977, 380096), (4176808, 4034962), (37110593, 4227657), (376208, 4082346), (4214376, 37311673), (376337, 4134145), (375252, 443767), (4202797, 43022019), (4130526, 201820), (4038502, 4080992), (4034966, 4058243), (432890, 4101478), (192279, 4206115), (442793, 4226354), (4311708, 376112), (31821, 4028942), (4101478, 36674752), (442793, 192279), (201820, 4058243), (45772914, 4270049), (261071, 45772914), (4174977, 376683), (437515, 4130526), (378743, 35626039), (37204336, 443916), (378416, 761559), (4058243, 4129519), (4134440, 761048), (4274025, 4180628), (441840, 4274025), (4201554, 432890), (4274964, 4102176), (4041283, 4202797), (764524, 765238), (3182674, 3172958), (35625722, 35625724), (4269871, 4221962), (4080992, 4162092), (761048, 4224419), (376683, 4269871), (376112, 4044392), (195771, 4178452), (195771, 4327944), (443767, 4174977), (31821, 201820), (4080992, 375252), (4178452, 4237068), (4044391, 4311708), (4274025, 433128), (4180628, 376337), (375252, 764524), (201820, 45766050), (761559, 37311253), (4130526, 4214376), (375252, 765692), (442793, 443767), (443767, 4102176), (443783, 4134145), (4134145, 4105639), (4082346, 4101478), (765063, 35625724), (378416, 4174977), (4180628, 31821), (4274025, 37204336), (4028942, 4129525), (4105639, 4044392), (441840, 4038502), (378416, 765238), (4044391, 4105639), (37311673, 4162239), (378743, 35626038), (201820, 195771), (4117771, 432890), (432795, 43530876), (4227657, 3172958), (443767, 4224419), (378416, 45757798), (43530876, 4294429), (4201554, 45757798), (35622016, 4129524), (4274025, 436670), (376683, 378743), (4274025, 4179873), (4174977, 4101478), (376337, 443730), (4274025, 376208), (201820, 201826), (376683, 37311254), (255919, 4117771), (31821, 4176808), (4180628, 4134440), (443916, 4077761), (4177703, 3182674), (380096, 4210128), (765238, 37311254), (761191, 765063), (436670, 4034964), (4176808, 4129524), (4274025, 35622958), (4130161, 35622016), (442793, 443730), (4274025, 443783), (4034964, 4029423), (442793, 4082346), (31821, 35622016), (443730, 4044391), (4226354, 4294429), (4162092, 378416), (4179873, 261071), (433128, 442793), (765692, 761559), (4102176, 35625722), (4082346, 4311708), (4077761, 4129524), (4274964, 761191), (201826, 4129519), (4206115, 45772914), (4038502, 4134440), (4080992, 4201554)}
```
<p>



</p>
</details>